### PR TITLE
Fix `run_xgettext.sh` code and make it work on osx

### PR DESCRIPTION
**Pull request by fabrixxm**
- run script thru `shellcheck` and fix code
- replace `readlink -f` with a function which works on osx too
- set flags to stop on errors

### CODE REVIEW

File: bin/run_xgettext.sh [hunk 1]
```diff
@@ -1,9 +1,20 @@
 #!/bin/bash
-FULLPATH=$(dirname $(readlink -f "$0"))
+set -eo pipefail
 
-if [ "$1" == "--help" -o "$1" == "-h" ]
+function resolve {
+	if [ "$(uname)" == "Darwin" ]
```
Comment (MrPetovan) [inline bin/run_xgettext.sh:5]: Uuuuuuugh.

File: bin/run_xgettext.sh [hunk 2]
```diff
@@ -61,6 +72,7 @@ KEYWORDS="-k -kt -ktt:1,2"
 echo "Extract strings to $OUTFILE.."
 rm "$OUTFILE"; touch "$OUTFILE"
 
+# shellcheck disable=SC2086  # FINDOPTS is meant to be splitted
```
Comment (MrPetovan) [inline bin/run_xgettext.sh]:
```suggestion
# shellcheck disable=SC2086  # $FINDOPTS is meant to be split
```

File: bin/run_xgettext.sh [hunk 3]
```diff
@@ -74,38 +86,39 @@ do
 	# On Windows, find still outputs the name of pruned folders
 	if [ ! -d "$file" ]
 	then
+		# shellcheck disable=SC2086  # KEYWORDS is meant to be splitted
```
Comment (MrPetovan) [inline bin/run_xgettext.sh]:
```suggestion
		# shellcheck disable=SC2086  # $KEYWORDS is meant to be split
```

File: bin/run_xgettext.sh [hunk 4]
```diff
@@ -74,38 +86,39 @@ do
...
 		xgettext $KEYWORDS -j -o "$OUTFILE" --from-code=UTF-8 "$file" || exit 1
-		sed -i "s/CHARSET/UTF-8/g" "$OUTFILE"
+		sed -i.bkp "s/CHARSET/UTF-8/g" "$OUTFILE"
```
Comment (MrPetovan) [inline bin/run_xgettext.sh:90]: What's the `.bkp` for?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by MrPetovan** [comment]
Also the CI pipeline apparently needs to be updated.

File: bin/run_xgettext.sh (hunk 1 above)
Comment (fabrixxm) [reply bin/run_xgettext.sh:5]: unfortunately, on linux `readlink -f` is the way to go, while `realpath` is deprecated (and looks like also in most of BSDs). But on osx `readlink` doesn't have `-f` flag, so the use of `realpath` is required. At least that's what I got looking the issue on "the internets"

File: bin/run_xgettext.sh (hunk 2 above)
Comment (fabrixxm) [reply bin/run_xgettext.sh]: Yeah.. ouf...

File: bin/run_xgettext.sh (hunk 4 above)
Comment (fabrixxm) [reply bin/run_xgettext.sh:90]: `sed -i [suffix]`  edit files in place and makes backup if extension supplied, but on osx the suffix is required.
I forgot to remove the backup at the end....

**Comment by fabrixxm** [comment]
> Also the CI pipeline apparently needs to be updated.

It's not a problem in pipeline. Looks like is a problem between `set -e` and `((count++))` on linux..
I'll look at it tomorrow...

**Comment by nupplaphil** [comment]
The overall question should be why we manually execute such scripts :X ;-) there are ways to create/sync translationfiles automatically, but that's just an impression for a future CI/CD script *gg*

File: bin/run_xgettext.sh (hunk 2 above)
Comment (annando) [reply bin/run_xgettext.sh]: Splitteted?

**Comment by MrPetovan** [comment]
After all, we do run it during CI so it'd be a matter of committing the result during CI run.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**